### PR TITLE
fix(interpreter): deepen StringToNumber; concentrate the WhiteSpace predicate

### PR DESCRIPTION
**Pull request by pmatos**
## What & why

This PR is the top recommendation from an architecture review run with the
`improve-codebase-architecture` skill (which surfaced deepening opportunities
across the interpreter). Of the candidates, `StringToNumber` was the highest
impact / lowest risk: a genuine **deepening** that also fixes three spec
conformance divergences, with a blast radius of one pure function plus five
call sites.

### The friction

`string_to_number` (the ECMAScript §7.1.4.1 ToNumber string path in
`helpers.rs`) was a shallow seam that leaked Rust's float grammar to every
ToNumber caller:

- it trimmed with Rust's `char::is_whitespace` (a *second*, wrong copy of the
  WhiteSpace concept — the canonical `is_ecma_whitespace` predicate lived in
  `builtins/string.rs`, unreachable to ToNumber's own module);
- it parsed `0x`/`0o`/`0b` literals via `i64::from_str_radix` (overflows); and
- it fell straight to `f64::parse` (accepts Rust's `inf`/`nan` spellings).

Three divergences from the spec (cross-checked with node) resulted:

| Input | Before | Spec / node |
|---|---|---|
| `Number(NEL + "1")` (U+0085) | `1` | `NaN` (NEL is not StrWhiteSpace) |
| `Number("0x10000000000000000")` | `NaN` | ~`2**64` (rounds, no overflow) |
| `Number("inf")` / `"+inf"` | `Infinity` | `NaN` (only `"Infinity"` is valid) |

### The deepening

- **Concentrate the WhiteSpace predicate.** `is_ecma_whitespace` moves into
  `helpers.rs` (the spec-conversions module), so `trim` / `parseInt` /
  `parseFloat` / ToNumber now all draw from **one** definition instead of
  ToNumber reinventing a wrong one. (locality: one predicate; leverage: 4 call sites)
- **Add `radix_digits_to_f64`.** It validates every non-decimal digit, parses the
  full value as an exact `BigUint`, and converts that exact integer to `f64`
  once. This avoids both fixed-width overflow and intermediate double rounding.
- **Reject `inf`/`infinity`/`nan`** before the final `parse::<f64>`, stopping the
  seam from leaking Rust's grammar.

## Tests (TDD: red -> green)

Written test-first against the ECMA-262 truth table (the independent source of
truth, cross-checked with node); the whitespace / hex-overflow / inf cases
failed on the old implementation before the fix.

- **Pure unit tests** for `string_to_number` in `helpers.rs` — whitespace set,
  inf/nan, non-decimal literals, exact rounding, decimals/empty.
- **JS-driven seam tests** at the public `Number(...)` interface in `tests.rs`.
- **`test262-extra/StringToNumber-whitespace-nondecimal-infinity.js`** — a
  spec-pattern test following the repo's convention, since test262 has no
  coverage for these negative edge cases.

## Validation

- `./scripts/lint.sh`
- 340 release unit tests plus the test262 smoke oracle
- release build
- 9/9 custom tests
- focused `built-ins/Number` test262: 680/680
- full default test262: 99,546/99,568; all 22 failures are in unrelated Intl
  DateTimeFormat/Temporal paths and reproduce unchanged with an exact
  `origin/main` binary, so this patch has zero regressions

## Disclosures

- **README unchanged.** These edge cases have no test262 coverage, so the tracked
  pass count does not move.
- **Scope.** This fixes the three named divergences, including correctly rounded
  non-decimal integers of arbitrary size. Pre-existing decimal-grammar behaviour
  rides the unchanged `parse::<f64>` path and is out of scope.

### CODE REVIEW

File: src/interpreter/helpers.rs [hunk 1]
```diff
@@ -70,39 +70,85 @@ pub(crate) fn to_number(val: &JsValue) -> f64 {
...
+// values ≤ 2^53; above that it is not guaranteed bit-identical to RoundMVResult.
+// Empty or invalid input → NaN.
+fn radix_digits_to_f64(digits: &str, radix: u32) -> f64 {
+    if digits.is_empty() {
+        return f64::NAN;
+    }
+    let mut acc = 0.0_f64;
+    for ch in digits.chars() {
+        match ch.to_digit(radix) {
+            Some(d) => acc = acc * f64::from(radix) + f64::from(d),
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/interpreter/helpers.rs]: Fixed in `7f2a516`.

Changed: `radix_digits_to_f64` now validates every radix digit, parses the full value as an exact `BigUint`, and converts that exact integer to `f64` once. Added regressions for both `0x6269e107215582e` and `0x200000000000011` at the helper, public `Number(...)`, and `test262-extra` seams; numeric separators remain rejected.

Validation: `./scripts/lint.sh`; 340 release unit tests plus the smoke oracle; release build; 9/9 custom tests; focused `Number` test262 680/680. The full test262 run was 99,546/99,568; its 22 Intl failures were reproduced unchanged with an exact `origin/main` binary, so they are baseline/environment noise unrelated to this fix.
